lib/omniauth/strategies/slack.rb

```diff
@@ -19,56 +19,37 @@ class Slack < OmniAuth::Strategies::OAuth2
         param_name: 'token'
       }
 
-      uid { raw_info['user_id'] }
+      # User ID is not guaranteed to be globally unique across all Slack users.
+      # The combination of user ID and team ID, on the other hand, is guaranteed
+      # to be globally unique.
+      uid { "#{user_identity['id']}-#{team_identity['id']}" }
 
       info do
         hash = {
-          nickname: raw_info['user'],
-          team: raw_info['team'],
-          user: raw_info['user'],
-          team_id: raw_info['team_id'],
-          user_id: raw_info['user_id']
+          name: user_identity['name'],
+          email: user_identity['email'],    # Requires the identity.email scope
+          image: user_identity['image_48'], # Requires the identity.avatar scope
```

Comment:
I tried to follow the specs provided by [Auth Hash Schema 1.0](https://github.com/intridea/omniauth/wiki/Auth-Hash-Schema#schema-10-and-later) where it’s stated that the image should be “roughly 50x50 pixel”.

All other sizes will still be available as extra info in the `raw_info` hash, provided you requested the `users:read` scope.

```
"raw_info": {
  "user_info": {
    "user": {
      "profile": {
        "image_24": "https:\/\/...",
        "image_32": "https:\/\/...",
        "image_48": "https:\/\/...",
        "image_72": "https:\/\/...",
        "image_192": "https:\/\/..."
```